dashboards/resource_lakeview_dashboard.go

```diff
@@ -0,0 +1,233 @@
+				if ok {
+					return true
+				}
+				_, new_json_hash, _ := readSerializedJsonContent(new, "")
+				// fmt.Println(new_json_hash, d.Get("md5").(string))
+				// fmt.Println(new, d.Get("serialized_dashboard").(string), old)
+				return d.Get("md5").(string) == new_json_hash && !d.Get("dashboard_change_detected").(bool)
+				// return !d.Get("dashboard_change_detected").(bool)
+			}
+			s["update_time"].Computed = true
+			s["warehouse_id"].Optional = false
+			s["warehouse_id"].Required = true
+
+			s["file_path"] = &schema.Schema{
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"serialized_dashboard"},
+			}
+			s["md5"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			}
+			s["dashboard_change_detected"] = &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			}
+			s["embed_credentials"] = &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			}
+			return s
+		})
+	return common.Resource{
+		Schema: s,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var new_dashboard dashboards.CreateDashboardRequest
+			common.DataToStructPointer(d, s, &new_dashboard)
+			content, md5Hash, err := readSerializedJsonContent(d.Get("serialized_dashboard").(string), d.Get("file_path").(string))
+			if err != nil {
+				return err
+			}
+			d.Set("md5", md5Hash)
+			new_dashboard.SerializedDashboard = content
+			created_dashboard, err := w.Lakeview.Create(ctx, new_dashboard)
+			if err != nil {
+				return err
+			}
+
+			time.Sleep(2 * time.Second)
+			dashboard := dashboards.GetDashboardRequest{
+				DashboardId: created_dashboard.DashboardId,
+			}
+			resp, err := w.Lakeview.Get(ctx, dashboard)
+			if err != nil {
+				return err
+			}
+			d.Set("etag", resp.Etag)
+
+			// Publish the dashboard
+			// fmt.Println("Publishing the dashboard", d.Get("embed_credentials").(bool))
+			_, err = w.Lakeview.Publish(ctx, dashboards.PublishRequest{
+				DashboardId:      created_dashboard.DashboardId,
+				WarehouseId:      d.Get("warehouse_id").(string),
+				EmbedCredentials: d.Get("embed_credentials").(bool),
+				ForceSendFields:  []string{"EmbedCredentials"},
+			})
+			if err != nil {
+				return err
+			}
+
+			d.SetId(created_dashboard.DashboardId)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			dashboard := dashboards.GetDashboardRequest{
+				DashboardId: d.Id(),
+			}
+			resp, err := w.Lakeview.Get(ctx, dashboard)
+			if err != nil {
+				return err
+			}
+			// fmt.Println(d.Get("md5").(string))
+			d.Set("dashboard_change_detected", (resp.Etag != d.Get("etag").(string)) || func() bool {
+				if v, ok := d.GetOk("file_path"); ok {
```

Comment:
THis will work, but I think we only need to detect remote changes here with the Etag. The DiffSuppress function above should take care of the local changes.

Reply:
I tried creating custom diff function for file as well. But somehow, checking md5 hash of file in read function works better. (now there is no custom diff for `file_path` field